## Lesson 6 ##

### Task 1 ###

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

#### Решение ####

In [1]:
import numpy as np

In [6]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])

ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])


In [11]:
cov = (zp * ks).mean() - zp.mean()* ks.mean()
cov

9157.839999999997

In [16]:
np.cov(zp, ks, ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [23]:
S_zp = (np.sum(((zp - zp.mean()) ** 2)) / len(zp)) ** (1/2)
S_zp

59.115480206118605

In [25]:
S_zp_np = np.std(zp)
S_zp_np

59.115480206118605

In [26]:
S_ks = (np.sum(((ks - ks.mean()) ** 2)) / len(ks)) ** (1/2)
S_ks

174.55340157098058

In [27]:
S_ks_np = np.std(ks)
S_ks_np

174.55340157098058

In [28]:
r = cov/(S_zp * S_ks)
r

0.8874900920739158

In [29]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Ответ: r = 0.88749009

### Task 2 ###

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95

#### Решение ####

In [30]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [31]:
S = x.std(ddof=1)
S

10.54566788359614

In [32]:
A_2 = (1 - 0.95) /2
A_2

0.025000000000000022

In [33]:
t_a2 = 2.262

In [37]:
X_1 = x.mean() - t_a2 * S/(len(x)**(1/2))
X_1

110.55660776308164

In [39]:
X_2 = x.mean() + t_a2 * S/(len(x)**(1/2))
X_2

125.64339223691834

Ответ: [110.56, 125.64]

### Task 3 ###

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

#### Решение ####

In [40]:
D = 25
S = D**(1/2)
S

5.0

In [41]:
n = 27
M = 174.2

In [42]:
A_2 = 0.975

In [43]:
Z_a2 = 1.96

In [44]:
X_1 = M - Z_a2 * S / (n**(1/2))
X_1

172.31398912064722

In [45]:
X_2 = M + Z_a2 * S / (n**(1/2))
X_2

176.08601087935276

Ответ: [172.31, 176.09]